_rules/heading-accessible-name-ffd0e9.md

```diff
@@ -0,0 +1,153 @@
+---
+id: ffd0e9
+name: Heading has accessible name
+rule_type: atomic
+description: |
+  This rule checks that each heading has an accessible name.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag20:2.4.6: # Headings and Labels (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any HTML element with the [semantic role][] of `heading` that is [included in the accessibility tree][].
+
+## Expectation
+
+Each test target has a non-empty (`""`) [accessible name][].
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility Support
+
+Some assistive technologies may hide headings with empty [accessible name][] from the users. This depends both on the user agent and how the [accessible name][] was computed (the [accessible name and description computation][] is not clear concerning which characters should be trimmed) and of the assistive technology itself. Hence, there are cases where the outcome of this rule is _failed_, but users of certain assistive technology and browser combinations will not experience an issue.
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Info and Relationships][usc131]
+- [Understanding Success Criterion 2.4.6: Headings and Labels][usc246]
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `h1` element has a non-empty [accessible name][].
+
+```html
+<h1>ACT rules</h1>
+```
+
+#### Passed Example 2
+
+This `div` element has a [semantic role][] of heading and a non-empty [accessible name][].
+
+```html
+<div role="heading" aria-level="1">ACT rules</div>
+```
+
+#### Passed Example 3
+
+This `h1` element has an non-empty [accessible name][] given by the `aria-labelledby` attribute.
+
+```html
+<span id="h-name">ACT rules</span>
+<h1 aria-labelledby="h-name"></h1>
+```
+
+#### Passed Example 4
+
+This `h1` element has an non-empty [accessible name][] given by the `alt` attribute of its content.
+
+```html
+<h1><img src="#" alt="ACT rules" /></h1>
+```
+
+#### Passed Example 5
+
+Even though this `h1` element is not [visible][], it is still [included in the accessibility tree][]. It has a non-empty [accessible name][]
+
+```html
+<h1 style="position: absolute; top: -9999px">ACT rules</h1>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `h1` element has an empty [accessible name][].
+
+```html
+<h1></h1>
```

Comment:
I don't think I would fail this during an actual audit. I have now read the debate that transformed this rule and the accessibility support section. But do we know of even a single mayor AT-UA combination that doesn't simply ignore this? A single example would ease my mind and convince me to fail this in future audits.

I have the same reservations about failed example 2.

Reply:
@ShadowBB will you be present on the next CG call (Nov. 28) to discuss this?

Reply:
What is different about failed example 3 that you don't have reservations about it? And if there are no other failed examples, should there be a rule?

What if failed example 1 were:
`<h1><span style="opacity: 0;" aria-hidden="true">Hidden content</span></h1>`

Reply:
In the test procedure of H42 Using h1-h6 to identify headings, the test procedure step 2 says to "check that heading markup is not used when content is not a heading."

https://www.w3.org/WAI/WCAG21/Techniques/html/H42

Reply:
@EmmaJP Now that you mention it, I am not sure I would fail failed example 3 either. Nor the example you posted. Unless someone can show me an example of a AT-UA combination that doesn't simply ignore this.

@DagfinnRomen You could interpret that as "There is no content. So it doesn't fail this technique." But even if it failed the technique. It still would not be a problem since it could succeed a different technique and still comply with this SC.

This is all a rather minor gripe. If the consensus of the A11Y community seems to be that we should fail this then I can get on board and fail this, but it would really help me explain this to clients if I had a single example of a AT-UA combination that actually mentions this header. And I can't find one.